traits/trait_set_object.py

```diff
@@ -263,35 +441,252 @@ def __reduce_ex__(self, protocol=None):
+        value : any
+            A value.
+
+        Returns
+        -------
+        self : set
+            The updated set.
+
+        Notes
+        -----
+        Parameters in the notification:
+            removed : set
+                A set containing the removed values.
+            added : set
+                An empty set.
+
+
+        """
         self.intersection_update(value)
         return self
 
     def __ixor__(self, value):
+        """ Return self ^= value.
+
+        Parameters
+        ----------
+        value : any
+            A value.
+
+        Returns
+        -------
+        self : set
+            The updated set.
+
+        Notes
+        -----
+        Parameters in the notification:
+            removed : set
+                A set containing the removed values.
+            added : set
+                A set containing the added values.
+
+        """
         self.symmetric_difference_update(value)
         return self
 
     def __isub__(self, value):
+        """ Return self-=value.
+
+        Parameters
+        ----------
+        value : any
+            A value.
+
+        Returns
+        -------
+        self : set
+            The updated set.
+
+        Notes
+        -----
+        Parameters in the notification:
+            removed : set
+                A set containing the removed values.
+            added : Undefined
+                Will be Undefined.
+
+        """
         self.difference_update(value)
         return self
+
+
+class TraitSetObject(TraitSet):
+    """ A specialization of TraitList with a default validator and notifier
+    which provide bug-for-bug compatibility with the TraitsListObject from
+    Traits versions before 6.0.
+
+    Parameters
+    ----------
+    trait : CTrait
+        The trait that the list has been assigned to.
+    object : HasTraits
+        The object the list belongs to.
+    name : str
+        The name of the trait on the object.
+    value : iterable
+        The initial value of the list.
+    notifiers : list
+        Additional notifiers for the list.
+
+    Attributes
+    ----------
+    trait : CTrait
+        The trait that the list has been assigned to.
+    object : HasTraits
+        The object the list belongs to.
+    name : str
+        The name of the trait on the object.
+    value : iterable
+        The initial value of the list.
+    """
+
+    def __init__(self, trait, object, name, value):
+
+        self.trait = trait
+        self.object = ref(object)
+        self.name = name
+        self.name_items = None
+        if trait.has_items:
+            self.name_items = name + "_items"
+
+        super().__init__(value, validator=self._validator,
+                         notifiers=[self.notifier])
+
+    def _validator(self, current_set, value):
+        """ Validates the value by calling the inner trait's validate method
+        and also ensures that the size of the list is within the specified
+        bounds.
```

Comment:
Please remove the mention of "ensures that the size of the list is within the specified bounds", likewise for the "Raises" section where this is mentioned again.

Reply:
Done.